src/main/java/org/wise/portal/presentation/web/controllers/author/project/ProjectAssetAPIController.java

```diff
@@ -116,7 +121,20 @@ private boolean isUserAllowedToUpload(User user, MultipartFile file) {
       allowedTypes += ","
           + appProperties.getProperty("trustedAuthorAllowedProjectAssetContentTypes");
     }
-    return allowedTypes.contains(file.getContentType());
+    return allowedTypes.contains(getRealMimeType(file));
+  }
+
+  public String getRealMimeType(MultipartFile file) {
+    AutoDetectParser parser = new AutoDetectParser();
+    Detector detector = parser.getDetector();
+    try {
+      Metadata metadata = new Metadata();
+      TikaInputStream stream = TikaInputStream.get(file.getInputStream());
+      org.apache.tika.mime.MediaType mediaType = detector.detect(stream, metadata);
+      return mediaType.toString();
+    } catch (IOException e) {
+      return MimeTypes.OCTET_STREAM;
+    }
```

Comment:
If we add OCTET_STREAM to allowedTypes in application.properties, could this allow someone to potentially upload a file that we do not support?

To handle this case, should this method throw the IOException instead of returning a default OCTET_STREAM, and catch the exception in isUserAllowedToUpload() method and return false there?